db/queries.py

```diff
@@ -1,46 +1,87 @@
 from db.models import Message, User, Chat
-from django.db.models import Q, Count, F
+from django.db.models import Q, F
 
 
 def get_messages_that_contain_word(word: str) -> list[Message]:
-    pass
+    return list(
+        Message.objects.filter(text__icontains=word)
+    )
 
 
 def get_untitled_chats() -> list[Chat]:
-    pass
+    return list(
+        Chat.objects.filter(title__contains="Untitled")
+    )
 
 
 def get_users_who_sent_messages_in_2015() -> list[str]:
-    pass
+    return list(
+        Message.objects.filter(sent__year=2015).values_list(
+            "user__first_name",
+            "user__last_name"
+        )
+    )
 
 
 def get_actual_chats() -> list[Chat]:
-    pass
+    return list(
+        Chat.objects.filter(message__sent__gte="2021-01-01")
+    )
 
 
 def get_messages_contain_authors_first_name() -> list[Message]:
-    pass
+    return list(
+        Message.objects.filter(text__contains=F("user__first_name"))
+    )
 
 
 def get_users_who_sent_messages_starts_with_m_or_a() -> list[User]:
-    pass
+    return list(
+        User.objects.filter(
+            Q(message__text__startswith="a")
+            | Q(message__text__startswith="m")
+        )
+    )
 
 
 def get_delivered_or_admin_messages() -> list[Message]:
-    pass
+    return list(
+        Message.objects.filter(
+            Q(is_delivered=True)
+            | Q(user__first_name__startswith="admin")
+        )
+    )
 
 
 def get_count_messages_sent_by_first_name(first_name: str) -> int:
-    pass
+    return Message.objects.filter(user__first_name=first_name).count()
 
 
 def get_top_users_by_number_of_the_messages() -> list[User]:
-    pass
+    users = User.objects.all()
+    for user in users:
+        user.num_messages = user.message_set.count()
+        user.save()
```

Comment:
By the way, this implementation is actually bad. It makes an `UPDATE` query to the database for every user.